### ISSUE TITLE
SocketAPI (new): Sync folder root does not have overlay if two syncs are configured

### ISSUE THREAD

**Issue by dragotin** [owncloud/client#4680]
### Expected behaviour
If user configures two sync connections and has both local folders in lets say /home/goofy/syncs both local sync folders should have an overlay icon.

### Actual behaviour
In my example, only one of the local directories shows an icon. The other has none. That is consistent over the linux nautilus and dolphin plugin.

### Steps to reproduce
1. Create two sync connections that have the local folder in the same directory. In my case, the syncs sync different servers.
2. Sync and check the overlay icons of the local root directories
3. Find that only one of the two has an ok icon.

**Comment by dragotin** [comment]
I tested again on Mac with the current nightly. In the beginning, same result: Only one local sync dir had a overlay.

A couple of minutes later, the second directory got a wrong overlay:
![bildschirmfoto 2016-04-14 um 11 42 03](https://cloud.githubusercontent.com/assets/1070214/14523934/44ed649e-0236-11e6-9f06-f90d66b5dbaa.png)

See that folder *marry* has a shared overlay which does not make sense for the root sync dir.

**Comment by dragotin** [comment]
Note: *marry* contains one shared directory.

**Comment by ckamm** [comment]
@dragotin I've not been able to reproduce with the nautilus script. I do see two icons and they match what gets sent via the socket API. I had ignored (hidden) files in the folder and the icon was yellow.

But when I removed the ignored files the icon became red. The log indicates the socketapi is... undecided:
```
Server response: UPDATE_VIEW:/home/kamm/ownCloud2/
Server response: STATUS:NOP:/home/kamm/ownCloud2/
Server response: STATUS:OK:/home/kamm/ownCloud2/
Server response: STATUS:NOP:/home/kamm/ownCloud2/
Server response: UPDATE_VIEW:/home/kamm/ownCloud2/
Server response: STATUS:NOP:/home/kamm/ownCloud2/
```

**Comment by ckamm** [comment]
It looks like `SFST::fileStatus("")` returns None. @jturcotte I'm not exactly sure what the correct behavior would be here. Just call lookupProblem("")` and return UpToDate if nothing is found?

**Comment by jturcotte** [comment]
By reading the code quickly it seems like the only way that it can return StatusNone is if the file is neither in the synfile vector, nor in the database. Is it expected in that case?

The status of parent folders will change as they get completed, that's probably why you see it passing to OK and then NONE.

The socket API now really just gives the value in the database, so is it possible that the permission for "mary" contains the shared flag?

**Comment by dragotin** [comment]
I'll try to debug that more

**Comment by dragotin** [comment]
A start: #4689

**Comment by dragotin** [comment]
this will be fixed with pr #4689

**Comment by mcastroSG** [comment]
Hi hi:

**Steps executed**

1. Create a new account connection to a server.
2. Define for this account a sync connection that has the local folder in the same directory that already defined.
3. Check that both folders shows a proper overlay icon.

**Results**

It´s working fine at OSX and Ubuntu.

Client OS: OS X El capitán 10.11.4
Client Version: Version 2.2.0beta1 (build 3297)

Client OS: Ubuntu 14.04 LTE
Client Version: Version 2.2.0nightly2016428 (build 1573)
